src/mooc/test/tv/codely/mooc/video/application/search/VideosSearcherShould.scala

```diff
@@ -16,4 +19,20 @@ final class VideosSearcherShould extends UnitTestCase with VideoRepositoryMock {
 
     searcher.all().futureValue shouldBe existingVideos
   }
+
+  "search an existing video" in {
+    val videoId = VideoIdMother.random
+    val videoTitle = VideoTitleMother.random
+    val videoDuration = VideoDurationMother.random
+    val videoCategory = VideoCategoryMother.random
+    val userId = UserIdMother.random
+    val existingVideo = Video(videoId, videoTitle, videoDuration, videoCategory, userId)
+
+    repositoryShouldSave(existingVideo)
```

Comment:
Why is saving in the search?

Reply:
In order to check the infraestructure in the acceptance test, and check an existing video is found, we save it first, so this test doesn't depend on another test to put an existing video in de Database.

Reply:
Since we're testing the application layer and mocking the database we don't need to insert before (we should do that in the acceptance tests) 😊

Reply:
Sorry, didn't realized of the "with VideoRepositoryMock", I saw the test in the /application/video namespace and thought it was an acceptance test. My bad. Thank you!